### ISSUE TITLE
Requires \<Esc> after second Given/Do blocks on Travis

### ISSUE THREAD

**Issue by christianrondeau** [junegunn/vader.vim#98]
Following up on #96,  I think found another issue (probably the last). 

This passes: https://travis-ci.org/christianrondeau/vim-base64/builds/170620930

This fails: 
https://travis-ci.org/christianrondeau/vim-base64/builds/170620115

The only difference is a `\<Esc>` at the beginning of the `Do` block that fixes it. It looks like there is either an operator pending, a visual selection or something that interferes with the `Given` block. I guess always doing an escape before executing the Do might work as a cheap fix.

**Comment by junegunn** [comment]
Another nasty Vim bug? Hmm. Can you give a simple script for reproducing the problem you described as you did in https://github.com/junegunn/vader.vim/issues/96#issuecomment-256015743?

**Comment by christianrondeau** [comment]
Yup, that was my intent :) it's somewhat painful to diagnose because it's only failing on Travis, but Vader is looking so good, it's worth it :) thanks, I'll report back when I figure it out.

**Comment by junegunn** [comment]
The problem should be reproducible with the old Vim installed on Travis CI environment (Ubuntu 12.04). So I would use Dockerfile like follows for testing it.

```dockerfile
# docker build -t vader-ubuntu-12.04 - < Dockerfile
# docker run -i -v tests:/root/tests -t vader-ubuntu-12.04
FROM ubuntu:12.04
RUN  apt-get update
RUN  apt-get install -y vim git curl
RUN  cd && git clone https://github.com/junegunn/vader.vim.git && \
     echo 'set rtp+=~/vader.vim' >> ~/.vimrc
CMD  cd && bash
```

**Comment by christianrondeau** [comment]
I'm mostly working on Termux (a Linux distribution for Android) so Docker is not an option, unless I'm at my home computer (which is a rare thing recently!). In the meantime, Travis is my best bet, but I'll certainly give it a try sometime! I'll keep you updated, thanks :)

**Comment by christianrondeau** [comment]
So I actually could setup the Docker machine, and it was way easier to drill down to the actual issue :) It might be possible to simplify it further, but that shows the issue pretty well with no non-vim code:

    Given (an encoded string):
      <value>QWJjMTIz</value>

    Do (visual decode):
      f>lvt<
    " Remove the `[v`]h and it works
      cAbc123\<esc>`[v`]h

    Expect (decoded string):
      <value>Abc123</value>

    Given (a string):
      <value>Abc123</value>

    Do (visual encode):
    " You can also add this, and the test passes
    "  \<Esc>
      f>lvt<
      cQWJjMTIz\<esc>

    Expect (encoded string):
      <value>QWJjMTIz</value>

The test output:

    (2/2) [ EXPECT] (X) encoded string
      - Expected:
          <value>QWJjMTIz</value>
      - Got:
          <value>Abc123</value>

So, removing this in the first `Do` block:

    `[v`]h

works, as well as adding `\<Esc>` in the *second* `Do` block. Something feel weird, I can't put my finger on it; it's like the visual selection affects the next test somehow.

A simple suggestion might be to force an "Escape" before starting a "Do" block to ensure there is no pending operators nor visual selection, but since it actually *works* in Vim 8, I guess there also may be something more going on.

Let me know if you have an idea for me to explore, and thanks again for the Docker suggestion!

**Comment by junegunn** [comment]
Thanks for looking into it. I just pushed the commit that does exactly what you suggested, and confirmed that it fixes the problem on old versions of Vim.

**Comment by justinmk** [comment]
`<C-\><C-N>` might be a bit better, it also is guaranteed to work in every mode. `<ESC>` may cause a beep and who knows what else.

**Comment by christianrondeau** [comment]
@junegunn I confirm that it works for me! You, sir, rock. Kudos from afar.

@justinmk I think you are correct, though this is not an issue for me specifically because A) I disable all error bells and B) I can't hear Travis from here :) I confirm that Esc does cause a beep, so that might be another issue resulting from this fix. I'll let @junegunn decide what to do with this, I think I bugged him enough for now :)
